src/snapshots/occlum/unionfs.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This unionfs file is used for occlum only
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::atomic::AtomicUsize;
+
+use anyhow::{anyhow, Result};
+use dircpy::CopyBuilder;
+use fs_extra;
+use fs_extra::dir;
+use nix::mount::MsFlags;
+
+use crate::snapshots::{MountPoint, Snapshotter};
+
+#[derive(Debug)]
+pub struct Unionfs {
+    pub data_dir: PathBuf,
+    pub index: AtomicUsize,
+}
+
+fn clear_path(mount_path: &Path) -> Result<()> {
+    let mut from_paths = Vec::new();
+    let paths = fs::read_dir(
+        mount_path
+            .to_str()
+            .ok_or(anyhow!("mount_path does not exist"))?,
+    )?;
+    for path in paths {
+        from_paths.push(path?.path());
+    }
+    fs_extra::remove_items(&from_paths)?;
+
+    Ok(())
+}
+
+fn create_dir(create_path: &PathBuf) -> Result<()> {
+    if !create_path.exists() {
+        fs::create_dir_all(create_path.as_path())?;
+    }
+
+    Ok(())
+}
+
+fn create_environment(mount_path: &Path) -> Result<()> {
+    let mut from_paths = Vec::new();
+    let mut copy_options = dir::CopyOptions::new();
+    copy_options.overwrite = true;
+
+    // copy the libs required by occlum to the mount path
+    from_paths.push(PathBuf::from("/lib64/ld-linux-x86-64.so.2"));
+    let mount_path_lib64 = mount_path;
+    let path_lib64 = mount_path_lib64.join("lib64");
+    create_dir(&path_lib64)?;
+
+    fs_extra::copy_items(&from_paths, &path_lib64, &copy_options)?;
+    from_paths.clear();
+
+    let mount_path_opt = mount_path;
+    let path_opt = mount_path_opt
+        .join("opt")
+        .join("occlum")
+        .join("glibc")
+        .join("lib");
+    create_dir(&path_opt)?;
+
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libc.so.6"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libdl.so.2"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libm.so.6"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libpthread.so.0"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libresolv.so.2"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/librt.so.1"));
```

Comment:
Could you use a loop for this to both reduce the number of lines but also only have a single occurence of `/opt/occlum` (could that ever change btw?)

Reply:
> Could you use a loop for this to both reduce the number of lines but also only have a single occurence of `/opt/occlum` (could that ever change btw?)

I could use a loop for this part.  The benefit is that only add an item in the string slice in the future if a new lib have to be added required by occlum.